Homework

Caffadras opened the pull request.

Caffadras committed:
Implemented StudentMap

Caffadras committed:
Bugfixes:
1. map size doubles after rehashing
2. NPE when trying to put new element in the bucket with existing elements

Caffadras committed:
Implemented tests and adjusted StudentMap.
Summary:
1. StudentMap now allows null key and values
2. Refactored to using Object.equals() to prevent NPE
3. containsKey() does not use get() method anymore as StudentMap now supports null values

Caffadras committed:
Generified StudentMap and wrote tests for it.
Summary:
1. GenericMap does not throw exception when called get() or remove() with key of a wrong type
2. Some refactoring

boaghivasile commented on src/main/java/com/endava/internship/collections/GenericMap.java:
```diff
@@ -0,0 +1,256 @@
+package com.endava.internship.collections;
+
+import java.util.*;
```
Avoid using wildcard imports, use only those which you need.

boaghivasile commented on src/main/java/com/endava/internship/collections/StudentMap.java:
```diff
@@ -1,78 +1,266 @@
-import java.util.Collection;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Avoid using wildcard imports, use only those which you need.

boaghivasile commented on src/test/java/com/endava/internship/collections/GenericMapTest.java:
```diff
@@ -0,0 +1,340 @@
+        );
+
+    }
+}
```
Small note, each file should end with an empty line to eliminate this red symbol.

boaghivasile commented on src/test/java/com/endava/internship/collections/StudentMapTest.java:
```diff
@@ -0,0 +1,271 @@
+        );
+
+    }
+}
```
Small note, each file should end with an empty line to eliminate this red symbol.

boaghivasile reviewed (approved):
Generally, your PR looks good to me, one minor thing to note, remove all the redundant comment notes.
Other than that, like the coverage on the tests.
![image](https://user-images.githubusercontent.com/43186373/190968758-333c06ce-4ffb-4e2d-b7cc-f59e6b4747ae.png)

LGTM!

Caffadras committed:
Resolved review comments.
1. Added new line at the end of each file
2. Removed redundant comments
3. Removed wildcard imports
